gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/BigQueryImplTest.java

```diff
@@ -0,0 +1,982 @@
+    String cursor = "cursor";
+    ImmutableList<JobInfo> jobList = ImmutableList.of(QUERY_JOB_WITH_PROJECT,
+        LOAD_JOB_WITH_PROJECT);
+    Tuple<String, Iterable<Job>> result =
+        Tuple.of(cursor, Iterables.transform(jobList, new Function<JobInfo, Job>() {
+          @Override
+          public Job apply(JobInfo jobInfo) {
+            return jobInfo.toPb();
+          }
+        }));
+    EasyMock.expect(bigqueryRpcMock.listJobs(JOB_LIST_OPTIONS)).andReturn(result);
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    Page<JobInfo> page = bigquery.listJobs(JOB_LIST_ALL_USERS, JOB_LIST_STATE_FILTER);
+    assertEquals(cursor, page.nextPageCursor());
+    assertArrayEquals(jobList.toArray(), Iterables.toArray(page.values(), JobInfo.class));
+  }
+
+  @Test
+  public void testListJobsWithSelectedFields() {
+    String cursor = "cursor";
+    Capture<Map<BigQueryRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    ImmutableList<JobInfo> jobList = ImmutableList.of(QUERY_JOB_WITH_PROJECT,
+        LOAD_JOB_WITH_PROJECT);
+    Tuple<String, Iterable<Job>> result =
+        Tuple.of(cursor, Iterables.transform(jobList, new Function<JobInfo, Job>() {
+          @Override
+          public Job apply(JobInfo jobInfo) {
+            return jobInfo.toPb();
+          }
+        }));
+    EasyMock.expect(bigqueryRpcMock.listJobs(capture(capturedOptions))).andReturn(result);
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    Page<JobInfo> page = bigquery.listJobs(JOB_LIST_OPTION_FIELD);
+    assertEquals(cursor, page.nextPageCursor());
+    assertArrayEquals(jobList.toArray(), Iterables.toArray(page.values(), JobInfo.class));
+    String selector = (String) capturedOptions.getValue().get(JOB_OPTION_FIELDS.rpcOption());
+    System.out.println(selector);
+    assertTrue(selector.contains("etag,jobs("));
+    assertTrue(selector.contains("configuration"));
+    assertTrue(selector.contains("jobReference"));
+    assertTrue(selector.contains("statistics"));
+    assertTrue(selector.contains("state,errorResult),nextPageToken"));
+    assertEquals(80, selector.length());
+  }
+
+  @Test
+  public void testCancelJob() {
+    EasyMock.expect(bigqueryRpcMock.cancel(JOB)).andReturn(true);
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    assertTrue(bigquery.cancel(JOB));
+  }
+
+  @Test
+  public void testCancelJobFromJobId() {
+    EasyMock.expect(bigqueryRpcMock.cancel(JOB)).andReturn(true);
+    EasyMock.replay(bigqueryRpcMock);
+    bigquery = options.service();
+    assertTrue(bigquery.cancel(JobId.of(PROJECT, JOB)));
+  }
+
+  @Test
+  public void testQueryRequest() {
+    JobId queryJob = JobId.of(PROJECT, JOB);
+    com.google.api.services.bigquery.model.QueryResponse responsePb =
+        new com.google.api.services.bigquery.model.QueryResponse()
+            .setJobReference(queryJob.toPb())
+            .setJobComplete(false);
```

Comment:
Do you think we should also test the "true" case to check that the response is being setup correctly in that case?